confident_metrics/metrics.py

```diff
@@ -0,0 +1,161 @@
+import os
+import re
+from threading import Lock
+from typing import Union
+
+from prometheus_client import start_http_server
+from prometheus_client.metrics import MetricWrapperBase
+
+
+_prometheus_server = None
+PROMETHEUS_HOST = os.getenv("PROMETHEUS_HOST", "0.0.0.0")
+PROMETHEUS_PORT = int(os.getenv("PROMETHEUS_PORT", "8000"))
+
+
+class PreciseFloat:
+    """A floating-point value.
+
+    It is possible to add numbers to it, and the result is guaranteed to be precise
+    thanks to Kahan summation algorithm. That addition operation is thread safe.
+
+    """
+
+    def __init__(self):
+        """Initialize a new instance of PreciseFloat."""
+        self._value = 0.0
+        self._aux = 0
+        self._lock = Lock()
+
+    def __iadd__(self, amount):
```

Comment:
```suggestion
    def __iadd__(self, amount: float) -> "PreciseFloat":
```
(not sure about `amount` type)

Let's add types everywhere as we usually do.

Reply:
do we need regular `__add__` method also?

Reply:
Not really in this case. We will only update it inplace. We wont be doing arithmetics between Counters, just updating the counters with integers.